src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -974,6 +853,55 @@ public WorkspacePanel(final MCreator mcreator) {
 		updateElementListRenderer();
 	}
 
+	/**
+	 * Adds a new section to this workspace as well as a vertical tab button on the left that switches
+	 * to the section panel when clicked.
+	 *
+	 * @param id              The unique identifier of the section used for reloading/filtering contained elements.
+	 * @param name            The name of the section shown in the workspace.
+	 * @param panel           The panel representing contents of the vertical tab being added.
+	 * @param available       If {@code false}, the section will be registered but not shown in the workspace.
+	 * @param switchCondition Determines whether clicking the tab button will actually switch the section.
+	 * @param <T>             The type of the section panel to be added.
+	 */
+	public <T extends JComponent & IReloadableFilterable> void addVerticalTab(String id, String name, T panel,

```

Comment:
Where would available be used?

It may also make sense to make an abstract class out of `JComponent & IReloadableFilterable` that also can provide methods that right now are implemented by BiPredicate called eg. WorkspacePanel

Reply:
Could be, only the name should be `WorkspaceSection` since the one you suggested is used by the class you commented on :)

Reply:
The one I commented on could be refactored to `WorkspacePanels`, it would fit better how all is called right now

Reply:
>Where would available be used?

This is included in documentation I guess:
>If {@code false}, the section will be registered but not shown in the workspace.

Reply:
This explains what it does, but my question is where would that be used in practice?

Reply:
Yes, this is what I meant

Reply:
This is part of the logic used by standard panels before this PR.

Reply:
Ok if it can't extend class, maybe make a bit another interface extending IReloadableFilterable called eg. IWorkspacePanel that also defines default methods for things that are right now parameters

Reply:
I refactored that part, now the question is if I may leave the `WorkspacePanel` renaming to you to perform after merging this PR?

Reply:
If I rename it here, I get like extra hundreds of changed lines and merge conflicts, so I don't think I should be the one to do that.

Reply:
I see, so now it will be abstract class / it is possible?

Reply:
Yes, it is already possible. Just a problem with name of `WorkspaceSectionPanel`. Would you rename the other class after this PR or do I name this one `WorkspaceSection`?

Reply:
I will do the renames later if needed